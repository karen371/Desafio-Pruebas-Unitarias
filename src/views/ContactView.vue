<!-- src/views/AboutView.vue -->
<template>
    <div>
      <h1>Contact</h1>
      <p>Esta es la pagina Contact</p>
    </div>
  </template>
  
  <script>
  export default {
    name: 'ContactView',
  };
  </script>s