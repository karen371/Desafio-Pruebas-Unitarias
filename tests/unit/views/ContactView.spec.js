// tests/unit/views/ContactView.spec.js
import { shallowMount, mount } from '@vue/test-utils';
import { createRouter, createWebHistory } from 'vue-router'
import ContactView from '@/views/ContactView.vue';

/**Prueba para verificar la renderizacion de la vista */
describe('ContactView', () => {
  test('Probando la existencia del componente o vista ContactView', async () => {
      const router = createRouter({
          history: createWebHistory(),
          routes: [{
              path: '/contact',
              name: 'ContactViewVue',
              component: ContactView
          }],
      });
      router.push('/contact');
      await router.isReady();

      const wrapper = mount(ContactView, {
          global: {
              plugins: [router]
          }
      });
      expect(wrapper.findComponent(ContactView).exists()).toBe(true);
  });
});


/**TEST PARA LA VISTA DE CONTACT Snapshots */
describe('Vista ContactView.vue', () => {
  test('ValidaciÃ³n de match con el snapshot', () => {
    const wrapper = shallowMount(ContactView);

    expect(wrapper.html()).toMatchSnapshot();
  });
});