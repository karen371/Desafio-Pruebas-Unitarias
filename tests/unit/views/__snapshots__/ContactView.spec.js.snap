// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vista ContactView.vue Validaci√≥n de match con el snapshot 1`] = `
<div>
  <h1>Contact</h1>
  <p>Esta es la pagina Contact</p>
</div>
`;
